client/landing/stepper/declarative-flow/internals/steps-repository/trial-acknowledge/hosting-trial-acknowledge.tsx

```diff
@@ -49,10 +50,14 @@ const CallToAction = ( { onStartTrialClick }: CallToActionProps ) => {
 const HostingTrialAcknowledgeInternal = ( { onStartTrialClick }: CallToActionProps ) => {
 	const { __ } = useI18n();
 	const plan = getPlan( PLAN_BUSINESS );
-
+	const isEligible = useSelector( isUserEligibleForFreeHostingTrial );
 	const planFeatures =
 		plan && 'getPlanCompareFeatures' in plan ? plan.getPlanCompareFeatures?.() ?? [] : [];
 
+	if ( ! isEligible ) {
+		return window.location.assign( '/setup/new-hosted-site' );
+	}
```

Comment:
Can we do that inside an `useLayoutEffect` instance? The reason is that every component should return a valid node, and this functional call doesn't do that.

Reply:
yeah this make sense

Reply:
I've just pushed an update to move the redirection to an effect callback.

Reply:
>  I was going to put it in the useLayoutEffect in the new-hosted-flow.ts file

Oh interesting! Yes that actually does look like a good place to put that. Similar to how we redirected logged out users. Like "you should never be here". The `useSideEffect` function even passes the current step name, so it's already all set up for us 👍